unreal_pak/src/file.rs

```diff
@@ -0,0 +1,151 @@
+use std::io::{Read, Seek, SeekFrom, Write};
+
+use flate2::{read::ZlibDecoder, write::ZlibEncoder, Compression};
+use sha1::{Digest, Sha1};
+
+use crate::error::PakError;
+use crate::header::{calculate_header_len, read_file_header, write_file_header, Block, Header};
+use crate::pakversion::PakVersion;
+use crate::CompressionMethod;
+
+/// Read a pak entry at the given offset in the
+///
+/// # Arguments
+///
+/// * `reader` - Anything that implements Read + Seek
+/// * `pak_version` - Version of the pak format used
+/// * `offset` - The offset of the start of the header of the file
+pub(crate) fn read_file<R>(
```

Comment:
This probably should be named `read_entry`. Is it also necessary to read header each time? This sounds like a big bottleneck.